packages/cal-api/src/controllers/calendarEntries/createCalendarEntry.ts

```diff
@@ -0,0 +1,52 @@
+import { NextFunction, Request, Response } from "express";
+import { updateRecurrenceRule } from "../../helpers/recurringEntriesHelpers";
+import { CalendarEntry, CalendarEntryType } from "../../models/calendarEntry";
+
+type CreateCalendarEntryInput =
+  | CreateNonRecurringCalendarEntryInput
+  | CreateRecurringCalendarEntryInput;
+
+type CreateNonRecurringCalendarEntryInput = {
+  calendarId: string;
+  creatorId: string;
+  title: string;
+  description?: string;
+  allDay: boolean;
+  recurring: false;
+  startTimeUtc: Date;
+  endTimeUtc: Date;
+};
+
+type CreateRecurringCalendarEntryInput = {
+  calendarId: string;
+  creatorId: string;
+  title: string;
+  description?: string;
+  allDay: boolean;
+  recurring: true;
+  startTimeUtc: Date;
+  endTimeUtc: Date;
+  frequency: string;
+  recurrenceEndsUtc: Date;
+  recurrencePattern: string;
+};
+
+export const createCalendarEntry = async (
+  req: Request<{}, {}, CreateCalendarEntryInput>,
+  res: Response,
+  _next: NextFunction
+) => {
+  try {
+    // ⚠️ when creating a calendar entry, leverage the req.params.id as calendarId
```

Comment:
On second thought, it seems unnecessary to specify the relationship of the resources within the route. Perhaps `POST /entries` will suffice. If so, then we should create a versioned endpoint so that it doesn't clash with the previous version of the endpoint.

Reply:
After discussing with @raym, either way is okay. So I will leave it as is.